internal/pkg/cli/job_init.go

```diff
@@ -323,6 +323,25 @@ func (o *initJobOpts) askSchedule() error {
 	return nil
 }
 
+func (o initJobOpts) legitimizePlatform() error {
+	detectedOs, detectedArch, err := o.dockerEngine.GetPlatform()
+	if err != nil {
+		return fmt.Errorf("get docker engine platform: %w", err)
+	}
+	detectedPlatform := aws.String(manifest.PlatformString(detectedOs, detectedArch))
```

Comment:
nit: What do you think of keeping`manifest.PlatformString` private, and instead here we can call `dockerengine.PlatformString` instead since the os and arch detected come from docker?